### PR TITLE
Add edit Role for user workload configuration

### PR DESCRIPTION
**Pull request by lilic**
This allows admin to grant this role to users to configure user workload monitoring stack via the user-workload-monitoring-config ConfigMap in the openshift-user-workload-monitoring namespace.

I choose the cluster-monitoring-operator task vs user workload as this role should be there for users to configure before the user workload stack is enabled.

cc @openshift/openshift-team-monitoring PTAL

* [x] I added CHANGELOG entry for this change.

### CODE REVIEW

File: jsonnet/cluster-monitoring-operator.jsonnet [hunk 1]
```diff
@@ -128,5 +128,23 @@ local metrics = import 'telemeter-client/metrics.jsonnet';
...
+      local configmapRule = policyRule.new() +
+                            policyRule.withApiGroups(['']) +
+                            policyRule.withResources([
+                              'configmaps',
+                            ]) +
+                            policyRule.withVerbs(['*']) +
+                            policyRule.withResourceNames(['user-workload-monitoring-config']);
+
+      role.new() +
+      role.mixin.metadata.withName('user-workload-monitoring-config-edit') +
```
Comment (s-urbaniak) [inline jsonnet/cluster-monitoring-operator.jsonnet:145]: nit: we specified `monitoring-workload-config-edit` in https://issues.redhat.com/browse/MON-984.

Comment (lilic) [reply jsonnet/cluster-monitoring-operator.jsonnet:145]: And I asked the comment on that issue why? :) As I am not sure that makes sense, as the role edits user workload no?

File: pkg/tasks/clustermonitoringoperator.go [hunk 2]
```diff
@@ -62,6 +62,16 @@ func (t *ClusterMonitoringOperatorTask) Run() error {
...
 	}
 
+	uwcr, err := t.factory.ClusterMonitoringEditUserWorkloadConfigRole()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkloadConfigEdit Role failed")
+	}
+
+	err = t.client.CreateOrUpdateRole(uwcr)
+	if err != nil {
+		return errors.Wrap(err, "reconciling UserWorkloadConfigEdit ClusterRole failed")
```
Comment (simonpasquier) [inline pkg/tasks/clustermonitoringoperator.go:72]:
```suggestion
		return errors.Wrap(err, "reconciling UserWorkloadConfigEdit Role failed")
```

File: jsonnet/cluster-monitoring-operator.jsonnet (hunk 1 above)
Comment (s-urbaniak) [reply jsonnet/cluster-monitoring-operator.jsonnet:145]: oh sorry, my jira notification iterations are in the mornings. I remember it was a conscious decision by PM @sichvoge as he does not want to leak the term "user workload monitoring" to end user APIs. Let me comment on the jira.

**Comment by lilic** [comment]
/retest

lease problem

/hold
until name is agreed on by PM and team lead

**Comment by lilic** [comment]
/hold

as discussion on name of role is again ongoing :)

**Comment by brancz** [comment]
/hold cancel

Seems like this was resolved.

**Pull request merged**